src/session.rs

```diff
@@ -535,6 +535,24 @@ impl<C> Session<C>
 where
     C: connect::Connect + Unpin + 'static + Clone + Send + Sync,
 {
+    fn new(
+        rx: mpsc::UnboundedReceiver<Task>,
+        client: hyper_util::client::legacy::Client<C, BoxBody<hyper::body::Bytes, Infallible>>,
+        wdb_url: url::Url,
+        session_id: &str,
```

Comment:
How about making this take `Option<impl Into<String>>` and then using `::new` for constructing the `Session` in both the arms of `setup_session`?

Reply:
Thanks, revised `::new` to accept `Option<impl Into<String>>`, ensuring it can handle both new and existing sessions seamlessly within `setup_session`.